protocol-units/bridge/shared/src/types.rs

```diff
@@ -135,13 +138,84 @@ impl From<Uint<256, 4>> for TimeLock {
 }
 
 #[derive(Deref, DerefMut, Debug, Clone, Copy, PartialEq, Eq)]
-pub struct Amount(pub u64);
+pub struct Amount(pub AssetType);
+/// The type of Asset being used
+#[derive(Clone, Debug, PartialEq, Eq, Copy)]
+pub enum AssetType {
+	/// Where the first tuple value is `Eth` and the second tuple value is `Weth`  
+	EthAndWeth((u64, u64)),
+	Moveth(u64),
+}
+
+#[derive(Error, Debug)]
+pub enum ConversionError {
+    #[error("Invalid conversion from AssetType to Uint")]
+    InvalidConversion,
+}
+
+impl TryFrom<AssetType> for Uint<256, 4> {
+    type Error = ConversionError;
+
+    fn try_from(value: AssetType) -> Result<Self, Self::Error> {
+        match value {
+            AssetType::EthAndWeth((eth_value, weth_value)) => {
+                // Example logic: combine the values or whatever makes sense in your context
+                let combined_value = eth_value as u128 + weth_value as u128;
+                Ok(Uint::from(combined_value))
+            }
+            AssetType::Moveth(value) => {
+                Ok(Uint::from(value as u128))
+            }
+            _ => Err(ConversionError::InvalidConversion), // Add more cases as needed
+        }
+    }
+}
+
+impl AddAssign for AssetType {
+	fn add_assign(&mut self, other: Self) {
+		match (self, other) {
+			(AssetType::Moveth(ref mut a), AssetType::Moveth(b)) => *a += b,
+			(AssetType::EthAndWeth((ref mut a, ref mut b)), AssetType::EthAndWeth((c, d))) => {
+				*a += c;
+				*b += d;
+			}
+			_ => panic!("Cannot add different AssetTypes"),
```

Comment:
Have this return an Result type instead of panicing as this would cause the program to crash. The Result type should be the same of the calling context then when you call `add_assign` you can append it with `?` to bubble up the error.